[BUG]: __sanitizer_annotate_contiguous_container definition in libcu++ conflicts with libc++

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Compile-time Error

### Component

libcu++

### Describe the bug

Attempting to compile host-side CUDA code with `clang -fsanitize=address` fails with conflicting function redeclaration:

```
In file included from /opt/compiler-explorer/cuda/12.6.1/include/cub/device/device_copy.cuh:42:
In file included from /opt/compiler-explorer/cuda/12.6.1/include/cub/detail/nvtx.cuh:49:
In file included from /opt/compiler-explorer/cuda/12.6.1/include/cub/detail/nvtx3.hpp:631:
In file included from /opt/compiler-explorer/clang-trunk-20241112/bin/../include/c++/v1/string:595:
/opt/compiler-explorer/clang-trunk-20241112/bin/../include/c++/v1/__debug_utils/sanitizers.h:24:1: error: __host__ function '__sanitizer_annotate_contiguous_container' cannot overload __host__ __device__ function '__sanitizer_annotate_contiguous_container'
   24 | __sanitizer_annotate_contiguous_container(const void*, const void*, const void*, const void*);
      | ^
/opt/compiler-explorer/cuda/12.6.1/include/cuda/std/detail/libcxx/include/__config:1434:1: note: previous declaration is here
 1434 | __sanitizer_annotate_contiguous_container(const void*, const void*, const void*, const void*);
      | ^
1 error generated when compiling for host.
Compiler returned: 1
```

### How to Reproduce

https://godbolt.org/z/sMfxKP7hM

### Expected behavior

Making `__sanitizer_annotate_contiguous_container` a `__host__` function avoids the problem.


### Reproduction link

https://godbolt.org/z/sMfxKP7hM

### Operating System

_No response_

### nvidia-smi output

_No response_

### NVCC version

_No response_

--- a/libcudacxx/include/cuda/std/detail/libcxx/include/__config
+++ b/libcudacxx/include/cuda/std/detail/libcxx/include/__config
@@ -677,11 +677,6 @@ typedef unsigned int char32_t;
 #    define _LIBCUDACXX_EXPLICIT_MOVE(x) (x)
 #  endif
 
-#  ifndef _LIBCUDACXX_HAS_NO_ASAN
-extern "C" _LIBCUDACXX_HIDE_FROM_ABI void
-__sanitizer_annotate_contiguous_container(const void*, const void*, const void*, const void*);
-#  endif
-
 // Thread API
 #  ifndef _LIBCUDACXX_HAS_THREAD_API_EXTERNAL
 #    if defined(_CCCL_COMPILER_NVRTC) || defined(__EMSCRIPTEN__)

